BUILDING.md

```diff
@@ -1,34 +1,68 @@
-See the [release](https://github.com/k3s-io/k3s/releases/latest) page for pre-built releases.
+**Note:** In case you are looking for the pre-built releases see the [release page](https://github.com/k3s-io/k3s/releases/latest).
+
+## Build k3s from source
+
+Before getting started, bear in mind that this repository includes all of
+Kubernetes history, so consider shallow clone it (`--depth 1`) to speed up the
+process.
 
-The clone will be much faster on this repo if you do
 ```bash
 git clone --depth 1 https://github.com/rancher/k3s.git
 ```
 
-This repo includes all of Kubernetes history so `--depth 1` will avoid most of that.
-
 The k3s build process requires some autogenerated code and remote artifacts that are not checked in to version control.
-To prepare these resources for your build environment, run:.
+To prepare these resources for your build environment, run:
+
 ```bash
 mkdir -p build/data && ./scripts/download && go generate
 ```
 
 To build the full release binary, you may now run `make`, which will create `./dist/artifacts/k3s`.
 
-To build the binaries using without running linting (ie; if you have uncommitted changes):
+To build the binaries using without running linting (i.e.: if you have uncommitted changes):
+
 ```bash
 SKIP_VALIDATE=true make
 ```
 
-If you make any changes to go.mod and want to update the vendored modules, you should run the following before running `make`:
+In case you make any changes to [go.mod](go.mod), you should run `go mod tidy` before running `make`.
```

Comment:
We can run `make deps` here.

Reply:
Sure, I can make that change. But `make deps` internally does `go mod vendor`, and I was under the impression that we are stirring away from `vendor/` directory. So, should I remove `go mod vendor` from the Makefile or add `vendor/` to the `.gitignore`?

Reply:
Let's get @luthermonson to weigh-in here.

Reply:
We should remove the `make deps` from the Makefile entirely, we do not want to go mod vendor anymore, and then the command is just a wrapper for `go mod tidy`

Reply:
I agree with @dereknola, however updating the Makefile or .gitignore might be outside of this PR scope,  and the documentation is still correct if we omit to use that make target. 

Either way, I'm open to do what the team sees proper.